internal/notifier/notifier.go

```diff
@@ -24,189 +28,327 @@ const (
 type NotifyFunc func(topic NotificationTopic, payload string)
 
 type Subscription struct {
-	creationTime time.Time
-	topic        NotificationTopic
 	notifyFunc   NotifyFunc
-
-	unlistenOnce *sync.Once
 	notifier     *Notifier
+	topic        NotificationTopic
+	unlistenOnce sync.Once
 }
 
-func (s *Subscription) Unlisten() {
+func (s *Subscription) Unlisten(ctx context.Context) {
 	s.unlistenOnce.Do(func() {
-		s.notifier.unlisten(s)
+		// Unlisten uses background context in case of cancellation.
+		if err := s.notifier.unlisten(context.Background(), s); err != nil { //nolint:contextcheck
```

Comment:
Is it safe to use a background context with no timeout here? I'm guessing not, because this underlying code:

https://github.com/riverqueue/river/blob/35ee45ee57e15b91eb1f722c4bf6ff69c003779d/riverdriver/riverpgxv5/river_pgx_v5_driver.go#L524-L530

Could hang forever if the context never times out and there isn't some other mechanism to force it to give up.

I get that you don't want to inherit from `ctx` because it may/will already be cancelled at the time this runs, but it probably needs at least _some_ timeout to make sure it doesn't hang forever.

Reply:
I'm still not sure that this is something that happens IRL, but in the interest of safety, just added timeouts to all listener operations.

Reply:
In the event I haven't recounted the outage at Mux that was caused by a lock taken within a transaction where an operation hung (likely some k8s/aws networking blip) and there was no txn or context timeout, remind me to do so 🤬